homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,155 @@
+    PALAZZETTI,
+)
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: PalazzettiConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Palazzetti climates based on a config entry."""
+    coordinator: PalazzettiDataUpdateCoordinator = entry.runtime_data
+    entities: list[PalazzettiClimateEntity] = []
+    if coordinator.client.connected:
+        entities.append(
+            PalazzettiClimateEntity(
+                coordinator=coordinator,
+            )
+        )
+        async_add_entities(entities)
+    else:
+        raise ConfigEntryNotReady
+
+
+class PalazzettiClimateEntity(
+    CoordinatorEntity[PalazzettiDataUpdateCoordinator], ClimateEntity
+):
+    """Defines a Palazzetti climate."""
+
+    _attr_has_entity_name = True
+    _attr_hvac_modes = []  # The available modes will be set when we know the current state
+    _attr_fan_modes = FAN_MODES
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(self, *, coordinator: PalazzettiDataUpdateCoordinator) -> None:
+        """Initialize Palazzetti climate."""
+        super().__init__(coordinator=coordinator)
+        client = coordinator.client
+        self._attr_unique_id = coordinator.config_entry.data[CONF_MAC]
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.config_entry.data[CONF_MAC])},
+            name=coordinator.config_entry.data[CONF_NAME],
+            manufacturer=PALAZZETTI,
+            sw_version=client.sw_version,
+            hw_version=client.hw_version,
+        )
+        self._attr_name = coordinator.config_entry.data[CONF_NAME]
+        self._attr_fan_modes = list(
+            str(range(client.fan_speed_min, client.fan_speed_max + 1))
+        )
+        if self.coordinator.client.has_fan_silent:
+            self._attr_fan_modes.insert(0, FAN_SILENT)
+        if self.coordinator.client.has_fan_high:
+            self._attr_fan_modes.append(FAN_HIGH)
+        if self.coordinator.client.has_fan_auto:
+            self._attr_fan_modes.append(FAN_AUTO)
+
+    @property
+    def available(self) -> bool:
+        """Is the entity available."""
+        return self.coordinator.client.connected
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Supported features."""
+        features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+        )
+        if self.coordinator.client.has_on_off_switch:
+            features |= ClimateEntityFeature.TURN_ON | ClimateEntityFeature.TURN_OFF
+        return features
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """List the hvac modes."""
+        if self.coordinator.client.has_on_off_switch:
+            return [HVACMode.OFF, HVACMode.HEAT]
+        return [self.hvac_mode]
```

Comment:
what does this mean? if the device has no switch it's either always on or off? if it's always off does this make sense?

Reply:
The on/off switch becomes unavaiables at times, during ignition and shut off cycles. An API call to turn on/off during that time will not raise an error but it will fail.

^ adding this as a comment for clarity

Reply:
The climate should then always return both modes and we should just handle the error gracefully, letting the user know what's going on (raising `HomeAssistantError`)

Reply:
But the API it letting me know that this action is not available at that moment. Why let the user take it and get back an error?

Reply:
Because then you can let the user know why it's not working instead of the option just disappearing. I believe we are usually don't change supported features at runtime.